pkg/sql/schemachanger/scbuild/internal/scbuildstmt/alter_table_alter_primary_key.go

```diff
@@ -550,9 +551,89 @@ func makeShardedDescriptor(b BuildCtx, t alterPrimaryKeySpec) *catpb.ShardedDesc
 func recreateAllSecondaryIndexes(
 	b BuildCtx, tbl *scpb.Table, newPrimaryIndex, sourcePrimaryIndex *scpb.PrimaryIndex,
 ) {
-	// TODO(postamar): implement in 23.1
-	// Nothing needs to be done because fallBackIfSecondaryIndexExists ensures
-	// that there are no secondary indexes by the time this function is called.
+	publicTableElts := b.QueryByID(tbl.TableID).Filter(publicTargetFilter)
+	// Generate all possible key suffix columns.
+	var inKeySuffix []indexColumnSpec
+	{
+		newPrimaryIndexElts := publicTableElts.Filter(hasIndexIDAttrFilter(newPrimaryIndex.IndexID))
+		scpb.ForEachIndexColumn(newPrimaryIndexElts, func(_ scpb.Status, _ scpb.TargetStatus, ic *scpb.IndexColumn) {
+			if ic.Kind == scpb.IndexColumn_KEY {
+				inKeySuffix = append(inKeySuffix, indexColumnSpec{
+					columnID:  ic.ColumnID,
+					kind:      scpb.IndexColumn_KEY_SUFFIX,
+					direction: ic.Direction,
+				})
+			}
+		})
+	}
+	// Recreate each secondary index.
+	scpb.ForEachSecondaryIndex(publicTableElts, func(_ scpb.Status, _ scpb.TargetStatus, idx *scpb.SecondaryIndex) {
+		out := makeIndexSpec(b, idx.TableID, idx.IndexID)
```

Comment:
nit: again for a few second I thought `out` means output heh. probably rename to `oldIndex` or `curIndex`. Similarly `inColumns` below probably can be `newColumns`.

Reply:
This actually conforms to an existing pattern elsewhere in the file, so I won't touch it. I agree the naming isn't great, though.